### PR TITLE
Back port credential update interval and hive.metastore.token.signatu…

### PR DESCRIPTION
**Pull request by sagarkapare**
…re configuration to 4.1

### CODE REVIEW

**Review by chtyim — APPROVED** [review]
LGTM

**Comment by anwar6953** [comment]
Can you link to the original JIRA or original PR?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedWorkflowProgramRunner.java [hunk 1]
```diff
@@ -174,9 +178,19 @@ protected ProgramController launch(Program program, ProgramOptions options,
     return createProgramController(controller, program.getId(), runId);
   }
 
-  private static YarnConfiguration createConfiguration(YarnConfiguration hConf) {
+  private static YarnConfiguration createConfiguration(YarnConfiguration hConf, CConfiguration cConf,
+                                                       TokenSecureStoreRenewer secureStoreRenewer) {
     YarnConfiguration configuration = new YarnConfiguration(hConf);
     configuration.setBoolean(HCONF_ATTR_CLUSTER_MODE, true);
+    configuration.set("hive.metastore.token.signature", HiveAuthFactory.HS2_CLIENT_TOKEN);
+    if (SecurityUtil.isKerberosEnabled(cConf)) {
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedWorkflowProgramRunner.java:186]: This is porting the fix added in https://github.com/caskdata/cdap/commit/30d28e9f3ce31b0b818712659c2a126a9086c354#diff-e715a8969f2a3f1cc4c42dd5ff219b39R102

File: cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/distributed/DistributedSparkProgramRunner.java [hunk 2]
```diff
@@ -115,6 +116,7 @@ private static YarnConfiguration createConfiguration(YarnConfiguration hConf, CC
                                                        TokenSecureStoreRenewer secureStoreRenewer) {
     YarnConfiguration configuration = new YarnConfiguration(hConf);
     configuration.setBoolean(SparkRuntimeContextConfig.HCONF_ATTR_CLUSTER_MODE, true);
+    configuration.set("hive.metastore.token.signature", HiveAuthFactory.HS2_CLIENT_TOKEN);
```
Comment (sagarkapare) [inline cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/distributed/DistributedSparkProgramRunner.java:119]: This was added as a support for spark2 on cdh here https://github.com/caskdata/cdap/commit/4b8b2eac210a65b5fdd36094cac7e64e3e48ac99#diff-9e51b61e5d3c6758ba9a00dcefd7a497R128

**Comment by sagarkapare** [comment]
Build is green - https://builds.cask.co/browse/CDAP-RUT1135-2. Merging.

**Pull request merged**